### PR TITLE
[Manure] Reimplements Anaerobic Lagoon

### LINKED ISSUE #2103: [Manure] Re-implement `AnaerobicLagoon` logic
Re-implement the `AnaerobicLagoon` logic. Use the "Developer Guidelines" in #2092.

### PR DESCRIPTION
**Pull request by ew3361zh**
Creates the newly refreshed Anaerobic Lagoon class.

### Context

Issue(s) closed by this pull request: closes #2103

### What

1. Establishes Storage child class Anaerobic Lagoon.
2. Refreshes current Anaerobic Lagoon emissions calculations and reporting according to refresh Storage structure.
3. Adds test files.
4. Updates input manure management json file with AnLag inputs from previous manure management json input file.

### Why

Part of phase 2 of manure refresh.

### How

1. Overrides parent `process_manure()` to facilitate emissions calculations, daily volume/water changes, and reporting.
2. Breaks up emissions calculations into helper functions to allow `process_manure()` child class function to be more easily testable. **This does create side-effects but it is noted in the docstrings of the helper functions and it seemed like a better decision than either creating and returning a new ManureStream object each time OR leaving the child `process_manure()` as too long and difficult to test easily. I think this also makes the individual emissions calculations and losses easier to follow and maintain than they were in the old setup.**
3. Adds reporting for anaerobic lagoon `storage_methane_burned`. I followed @allisterakun's model here but decided not to move it to the parent class since it won't be used by MOST of the storage child classes and could potentially be used for further special reporting for AnLag.
4. Makes same changes to Storage parent class as in #2285.
5. Updates input refreshed manure management json file as noted in What section above.

### Test plan

Unit testing fully covering new class.

### CODE REVIEW

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO: float = 9.25
+"""
+The mass conversion factor from methane to methane and carbon dioxide emitted from stored manure, based on a molar
+ratio of 1:3 (methane : carbon dioxide).
+"""
+
+SLURRY_MANURE_DENSITY = 990
+"""The density of slurry manure (kg/:math:`m^3`)."""
+
+STORAGE_HSC = 4.1
```
Comment (elle-andreen) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py:19]: Copying this from the slurry storage PR - Could we please rename this constant to "STORAGE_RESISTANCE" & update all its uses across all files? The name was incorrect in the existing manure module file

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+"""Housing specific constant for manure storage (s/m)."""
```
Comment (elle-andreen) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
"""Resistance value utilized in calculation of ammonia emission from manure stored in slurry storage outdoor, slurry storage underfloor, or anaerobic lagoon (s/m)"""
```

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py (hunk 1 above)
Comment (ew3361zh) [reply RUFAS/biophysical/manure/storage/anaerobic_lagoon.py:19]: Will hold off on this change in alignment with other storage child classes so that we can be sure to find all instances of it and move them to the base storage class.

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+        function_name = self.process_manure.__name__
+        self._report_processor_output(
+            "methane", total_storage_methane, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+        self._report_processor_output(
+            "methane_burned", storage_methane_burned, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
+        )
+        self._report_processor_output(
+            "ammonia", storage_ammonia, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
```
Comment (allisterakun) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
            "ammonia_N", storage_ammonia, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
```

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+        )
+        self._report_processor_output(
+            "nitrous_oxide", nitrous_oxide_emissions, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
```
Comment (allisterakun) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
            "nitrous_oxide_N", nitrous_oxide_emissions, function_name, MeasurementUnits.KILOGRAMS, time.simulation_day
```

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        )
```
Comment (allisterakun) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]: @elle-andreen when reporting these emission outputs, do we want to have them as "storage_methane" or just "methane"

**Review by allisterakun — APPROVED** [review]
LGTM! Just a few comments/discussion about the reported variable name. Just wanted it to be consistent across all Storages

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py (hunk 5 above)
Comment (elle-andreen) [reply RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]: @allisterakun I think we can continue to use "storage_methane". We can always change it if we decide "storage_" is too redundant but I think this specification in the variable name is helpful for output filter purposes

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+        total_methane = (
+            storage_methane_from_degradable_volatile_solids + storage_methane_from_non_degradable_volatile_solids
+        )
+        storage_methane_burned = 0.0
+        if self._cover == StorageCover.COVER_AND_FLARE:
+            storage_methane_burned, adjusted = self._calculate_cover_and_flare_methane(total_methane)
+            total_methane = adjusted
+
+        carbon_loss = total_methane * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```
Comment (elle-andreen) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
        mass_loss = total_methane * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```
This is just me being nitpicky, but I just want to modify this to avoid confusion because this value is not just carbon loss - the H and O (and probably other elements in reality) that are lost when CO2 and CH4 are produced in anaerobic fermentation are also included 💩

File: RUFAS/biophysical/manure/storage/anaerobic_lagoon.py [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+        self._manure_to_process.total_solids = max(0.0, self._manure_to_process.total_solids - carbon_loss)
```
Comment (elle-andreen) [inline RUFAS/biophysical/manure/storage/anaerobic_lagoon.py]:
```suggestion
        self._manure_to_process.total_solids = max(0.0, self._manure_to_process.total_solids - mass_loss)
```

File: tests/test_biophysical/test_manure/test_storage/test_anaerobic_lagoon.py [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+    )
+    anaerobic_lagoon._manure_to_process = stored_manure
+
+    mocker.patch.object(anaerobic_lagoon, "_calculate_methane_emissions", side_effect=[2.0, 1.0])
+
+    total, burned = anaerobic_lagoon._apply_methane_emissions(manure_temp)
+
+    expected_total = expected_total
+    expected_burned = expected_burned
+    carbon_loss = expected_total * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```
Comment (elle-andreen) [inline tests/test_biophysical/test_manure/test_storage/test_anaerobic_lagoon.py]:
```suggestion
    mass_loss = expected_total * METHANE_TO_METHANE_CARBON_DIOXIDE_RATIO
```

File: tests/test_biophysical/test_manure/test_storage/test_anaerobic_lagoon.py [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+    assert total == expected_total
+    assert burned == expected_burned
+    assert stored_manure.total_solids == pytest.approx(35.0 - carbon_loss, rel=1e-6)
```
Comment (elle-andreen) [inline tests/test_biophysical/test_manure/test_storage/test_anaerobic_lagoon.py]:
```suggestion
    assert stored_manure.total_solids == pytest.approx(35.0 - mass_loss, rel=1e-6)
```

File: input/data/manure/refreshed_manure_management.json [hunk 10]
```diff
@@ -35,5 +35,23 @@
...
+            "nitrogen_removal_efficiency_for_treatment": 0.0,
+            "total_ammoniacal_nitrogen_removal_efficiency_for_treatment": 0.0,
+            "phosphorus_removal_efficiency_for_treatment": 0.0,
+            "potassium_removal_efficiency_for_treatment": 0.0,
+            "hydraulic_retention_time": 365,
+            "sludge_accumulation_period": 10.0,
+            "sludge_accumulation_volume_fraction": 0.00,
+            "storage_time_period": 365,
+            "freeboard_input": 0.3048,
+            "manure_cover": "cover"
```
Comment (elle-andreen) [inline input/data/manure/refreshed_manure_management.json]: The required inputs for anaerobic lagoon are: 
        "name": "lagoon_1",
        "type": "anaerobic lagoon",
        "capacity": null,
        "cover": "no cover",
        "storage_time_period": 365,
        "surface_area": null

The removal efficiencies and other values can be removed! I tentatively set capacity to null for now - as I said in Allister's slurry storage PR, I think this value is not being used yet and having a value there could be confusing to users

**Review by elle-andreen — APPROVED** [review]
Left a few final comments/suggestions but I think this looks great! Thanks Niko!
